lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,148 @@ def post(self, request, *args, **kwargs):
+                or provide string 'all' to receive all blocks of the course.
+            requested_field (list): Optional list of names of additional fields to return for each block.
+                Supported fields can be found in transformers.SUPPORTED_FIELDS.
+            block_counts (list): Optional list of names of block types for which an aggregated count
+                of blocks is returned.
+            student_view_data (list): Optional list of names of block types for
+                which student_view_data is returned.
+            block_types_filter (list): Filter by block types:
+                'video', 'discussion', 'html', 'chapter', 'sequential', 'vertical'.
+            return_type (list, dict): Optional list or dictionary of block's fields based on 'return_type'.
+
+    **Response example**
+
+        Body consists of the following fields:
+
+        root: (str) The ID of the root node of the requested course block structure.\
+        blocks: (dict) A dictionary or list, based on the value of the
+            "return_type" parameter. Maps block usage IDs to a collection of
+            information about each block. Each block contains the following
+            fields.
+
+        id: (str) The Course's id (Course Run key)
+        name: (str) The course's name
+        number: (str) The course's number
+        org: (str) The course's organisation
+        start: (str) Date the course begins, in ISO 8601 notation
+        start_display: (str) Readably formatted start of the course
+        start_type: (str) Hint describing how `start_display` is set. One of:
+            * `"string"`: manually set by the course author
+            * `"timestamp"`: generated from the `start` timestamp
+            * `"empty"`: no start date is specified
+        end: (str) Date the course ends, in ISO 8601 notation
+        media: (dict) An object that contains named media items. Included here:
+            * course_image: An image to show for the course.  Represented
+              as an object with the following fields:
+                * uri: The location of the image
+        certificate: (dict) Information about the user's earned certificate in the course.
+            Included here:
+                * uri: The location of the user's certificate
+        is_self_paced: (bool) Indicates if the course is self paced
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent or the username was not provided
+        * 401 unauthorized, the provided access token has expired and is no longer valid
+          for an authenticated request.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    def get_certificate(self, request, course_id):
+        """Returns the information about the user's certificate in the course."""
```

Comment:
Given that this is a public function, can you document the input parameters and the format of the output data?

Reply:
Yes, sure

Reply:
input and output docs still missing.